pkg/collector/worker/worker.go

```diff
@@ -212,6 +225,13 @@ func (w *Worker) Run() {
 	log.Debugf("Runner %d, worker %d: Finished processing checks.", w.runnerID, w.ID)
 }
 
+func (w *Worker) shouldRunIntegrationInstance(check check.Check) bool {
+	if w.haAgent.Enabled() && w.haAgent.IsHaIntegration(check.String()) {
+		return w.haAgent.IsLeader()
+	}
+	return true
```

Comment:
Would it make sense to encapsulate this logic in the HA agent component?

```suggestion
	return w.haAgent.ShouldRun(check)
```

Reply:
Good idea, addressed here: https://github.com/DataDog/datadog-agent/pull/31186/commits/5ed60bbfe99f9fa395e9f6cd87f830836a7d29a9